tests/integrated/test-dataiterator/runtest

```diff
@@ -6,59 +6,31 @@
 
 from __future__ import print_function
 try:
-  from builtins import str
+    from builtins import str
 except:
-  pass
+    pass
 from boututils.run_wrapper import shell, shell_safe, launch_safe, getmpirun
 from boutdata.collect import collect
 from sys import stdout, exit
 
 MPIRUN=getmpirun()
 
 print("Making DataIterator test")
-s, out = launch_safe("make > make.log",pipe=True)
-if s:
-  print("Failed to run make")
-  exit(s);
+shell_safe("make > make.log",pipe=True)
 
-flags = [""]
-#, "nsys=2", "nsys=5 periodic", "nsys=7 n=10"]
-
-code = 0 # Return code
 for nproc in [1,2,4]:
-  for mproc in [1,2,4,8]:
-    cmd = "./test_dataiterator"
+    for mproc in [1,2,4,8]:
+        cmd = "./test_dataiterator"
+
+        print("   %d processors...." % (nproc))
 
-    print("   %d processors...." % (nproc))
-    r = 0
-    for f in flags:
         stdout.write("\tflags '"+f+"' ... ")
```

Comment:
`f` will not be defined on the first iteration and on subsequent iterations it will be a (closed) file object.